### PR TITLE
[Merged by Bors] - chore: update SHAs, forward-port

### PR DESCRIPTION
**Pull request by urkud**
Also fix some module docstrings.

* [`data.part`@`ee0c179cd3c8a45aa5bffbf1b41d8dbede452865`..`80c43012d26f63026d362c3aba28f3c3bafb07e6`](https://leanprover-community.github.io/mathlib-port-status/file/data/part?range=ee0c179cd3c8a45aa5bffbf1b41d8dbede452865..80c43012d26f63026d362c3aba28f3c3bafb07e6)
* [`order.bounds.basic`@`aba57d4d3dae35460225919dcd82fe91355162f9`..`3310acfa9787aa171db6d4cba3945f6f275fe9f2`](https://leanprover-community.github.io/mathlib-port-status/file/order/bounds/basic?range=aba57d4d3dae35460225919dcd82fe91355162f9..3310acfa9787aa171db6d4cba3945f6f275fe9f2)
* [`data.list.of_fn`@`fd838fdf07a83ca89fb66d30bebf6f0e02908c3f`..`bf27744463e9620ca4e4ebe951fe83530ae6949b`](https://leanprover-community.github.io/mathlib-port-status/file/data/list/of_fn?range=fd838fdf07a83ca89fb66d30bebf6f0e02908c3f..bf27744463e9620ca4e4ebe951fe83530ae6949b)
* [`data.set.finite`@`1126441d6bccf98c81214a0780c73d499f6721fe`..`1f0096e6caa61e9c849ec2adbd227e960e9dff58`](https://leanprover-community.github.io/mathlib-port-status/file/data/set/finite?range=1126441d6bccf98c81214a0780c73d499f6721fe..1f0096e6caa61e9c849ec2adbd227e960e9dff58)
* [`group_theory.group_action.opposite`@`fc2ed6f838ce7c9b7c7171e58d78eaf7b438fb0e`..`4330aae21f538b862f8aead371cfb6ee556398f1`](https://leanprover-community.github.io/mathlib-port-status/file/group_theory/group_action/opposite?range=fc2ed6f838ce7c9b7c7171e58d78eaf7b438fb0e..4330aae21f538b862f8aead371cfb6ee556398f1)
* [`data.list.basic`@`cf9386b56953fb40904843af98b7a80757bbe7f9`..`1447cae870f372074e480de1acbeb51de0077698`](https://leanprover-community.github.io/mathlib-port-status/file/data/list/basic?range=cf9386b56953fb40904843af98b7a80757bbe7f9..1447cae870f372074e480de1acbeb51de0077698)
* [`data.list.infix`@`6d0adfa76594f304b4650d098273d4366edeb61b`..`26f081a2fb920140ed5bc5cc5344e84bcc7cb2b2`](https://leanprover-community.github.io/mathlib-port-status/file/data/list/infix?range=6d0adfa76594f304b4650d098273d4366edeb61b..26f081a2fb920140ed5bc5cc5344e84bcc7cb2b2)
* [`set_theory.lists`@`9003f28797c0664a49e4179487267c494477d853`..`497d1e06409995dd8ec95301fa8d8f3480187f4c`](https://leanprover-community.github.io/mathlib-port-status/file/set_theory/lists?range=9003f28797c0664a49e4179487267c494477d853..497d1e06409995dd8ec95301fa8d8f3480187f4c)
* [`data.nat.choose.dvd`@`207cfac9fcd06138865b5d04f7091e46d9320432`..`966e0cf0685c9cedf8a3283ac69eef4d5f2eaca2`](https://leanprover-community.github.io/mathlib-port-status/file/data/nat/choose/dvd?range=207cfac9fcd06138865b5d04f7091e46d9320432..966e0cf0685c9cedf8a3283ac69eef4d5f2eaca2)

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

File: Mathlib/GroupTheory/GroupAction/Opposite.lean [hunk 1]
```diff
@@ -126,6 +126,7 @@ instance Semigroup.opposite_smulCommClass' [Semigroup α] : SMulCommClass α α
 #align semigroup.opposite_smul_comm_class' Semigroup.opposite_smulCommClass'
 #align add_semigroup.opposite_vadd_comm_class' AddSemigroup.opposite_vaddCommClass'
 
+@[to_additive]
```
[the remark this answers is not in the document]
Comment (eric-wieser) [reply Mathlib/GroupTheory/GroupAction/Opposite.lean:129]: Also, this PR forgot to add the `#align`.

File: Mathlib/Data/List/Infix.lean [hunk 2]
```diff
@@ -343,6 +343,20 @@ theorem mem_of_mem_take (h : a ∈ l.take n) : a ∈ l :=
...
+  _ = _ := take_append_drop _ _
+#align list.slice_sublist List.dropSlice_sublist
+
+lemma dropSlice_subset (n m : ℕ) (l : List α) : l.dropSlice n m ⊆ l :=
+  (dropSlice_sublist n m l).subset
+#align list.slice_subset List.dropSlice_subset
+
+lemma mem_of_mem_dropSlice {n m : ℕ} {l : List α} {a : α} (h : a ∈ l.dropSlice n m) : a ∈ l :=
+  dropSlice_subset n m l h
+#align list.mem_of_mem_slice List.mem_of_mem_dropSlice
```
Comment (eric-wieser) [inline Mathlib/Data/List/Infix.lean:358]: I PRd these already as #2049...

File: Mathlib/Data/List/OfFn.lean [hunk 3]
```diff
@@ -242,10 +241,16 @@ theorem ofFn_fin_repeat {m} (a : Fin m → α) (n : ℕ) :
     Nat.add_mul_mod_self_right, Nat.mod_eq_of_lt (Fin.is_lt _)]
 #align list.of_fn_fin_repeat List.ofFn_fin_repeat
 
+@[simp]
+theorem pairwise_ofFn {R : α → α → Prop} {n} {f : Fin n → α} :
+    (ofFn f).Pairwise R ↔ ∀ ⦃i j⦄, i < j → R (f i) (f j) := by
+  simp only [pairwise_iff_get, (Fin.cast (length_ofFn f)).surjective.forall, get_ofFn,
+    OrderIso.lt_iff_lt]
+#align list.pairwise_of_fn List.pairwise_ofFn
+
```
Comment (YaelDillies) [inline Mathlib/Data/List/OfFn.lean:250]: This was #1762...

**Pull request closed without merging**